test/test-functions

```diff
@@ -2448,6 +2399,12 @@ inst_binary() {
         # by ldd attempting to use the unprefixed RPATH.
         [[ "$line" =~ libsystemd.*\ not\ found ]] && continue
 
+        # We're built with ASan and the target binary loads one of the systemd's
+        # DSOs, so we need to tweak the environment before executing the binary
+        if get_bool "$IS_BUILT_WITH_ASAN" && [[ "$line" =~ /(libudev|libsystemd).so ]]; then
```

Comment:
I think it should look for all shared libraries `systemd` ships
```
$ find -name '*.so'
./build/libnss_systemd.so
./build/libcryptsetup-token-systemd-fido2.so
./build/libcryptsetup-token-systemd-pkcs11.so
./build/libnss_mymachines.so
./build/pam_systemd_home.so
./build/libudev.so
./build/libcryptsetup-token-systemd-tpm2.so
./build/libnss_resolve.so
./build/libsystemd.so
./build/src/shared/libsystemd-shared-251.so
./build/src/core/libsystemd-core-251.so
./build/libnss_myhostname.so
./build/pam_systemd.so
```
Without them it wouldn't be possible to log in in "interactive" mode for example:
```sh
H login: root
==69==ASan runtime does not come first in initial library list; you should either link runtime to your application or manually preload it with LD_PRELOAD.
```

Reply:
Though I suspect there `dlopen` is used to load the pam modules so the script can't figure it out with `ldd`

Reply:
Oh well, I tried several different *possible solutions*, and eventually just changed the regex to `'/(libudev|libsystemd.*|.+[\-_]systemd([\-_].+)?|libnss_(mymachines|myhostname|resolve)).so'` to keep my sanity, since apart from just hunting down all the systemd DSOs (which might be in several places, as we don't have to install a compiled systemd), there's also issue with various formats of the lines from `ldd` (sometimes it says something like `libsystemd... not found`, since the test libsystemd version is different from the host's one, etc.)

As for the interactive session, wrapping `login` unconditionally helped, but the terminal input with that is... very weird, I need to look into more.

Reply:
@mrc0mmand I think it would make sense to roll back to the first version of the PR for now and bring back `LD_PRELOAD` and the suppression file and create a bunch of wrappers manually. Given that the tests didn't fail anywhere it appears the pam and nss modules aren't covered by the testsuite so it's hard to tell what else can be broken. I think it can be revisited once those libraries are covered at least partially.

Reply:
I wonder if it would suffice to just drop the last two commits, and keep the automatic wrapper creation, but with `LD_PRELOAD=` and suppression file in place (just so we don't have to maintain the list manually, apart from the exceptions).

Reply:
Interesting. Turns out the "weird tty behavior" happens on the latest main as well, at least on my machine, so it's possible my setup is somehow borked, *sigh*. With wrapping `login` it seems to work as expected. I'll repush the previous version with the `login` binary and regex tweaked, just to see if CIs are going to complain.